From f007a8feba89931c1df27454ebfbd9f11c3e5194 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stephan=20D=C3=BCnner?= <stephan.duenner@bytesatwork.ch>
Date: Thu, 20 Jan 2022 09:00:38 +0100
Subject: [PATCH] WA: spi_nor: is25wp256 allow only standard read

Work-around for current HW.
Fixes following SPI NOR problem:

u-boot=> sf probe
fsl_fspi_nor: fail to find cmd 0x35 from lut
Failed to initialize SPI flash at 0:0 (error -22)
---
 drivers/mtd/spi/spi-nor-ids.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 1fd2adeeaf..0667f429cd 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -140,7 +140,7 @@ const struct flash_info spi_nor_ids[] = {
 	{ INFO("is25wp128",  0x9d7018, 0, 64 * 1024, 256,
 			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ INFO("is25wp256",  0x9d7019, 0, 64 * 1024, 512,
-			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+			SECT_4K) },
 #endif
 #ifdef CONFIG_SPI_FLASH_MACRONIX	/* MACRONIX */
 	/* Macronix */
-- 
2.30.2

